src/simulation/SimulationManager.java

```diff
@@ -225,20 +225,21 @@ public void performReverseStep(int steps) {
         int initialPos = statePath.getPosition();
         StateNode targetNode = statePath.getNode(initialPos < steps ? 0 : initialPos - steps);
 
-        // Decrement the position by (steps + 1) and then, if we're not past the beginning,
-        // step forward by 1, resulting in overall decrementing by (steps). It is done this way
-        // because the forward step gives us the A4Solution for a node, simplifyng the implementation
-        // for the `alt` command.
-        statePath.decrementPosition(steps + 1, traceMode);
         if (initialPos <= steps) {
-            return;
+            setToInit();
+        } else {
+            // Perform a "0-step" here to set the internal state properly for an alternate path
```

Comment:
A "0-step" is not being performed here. We only perform a "0-step" in `setToInit` or when the model is first loaded. Please keep the original comment here.

Reply:
In the comment, I was referring to a step with a delta of 0 – but I'll remove this terminology to prevent confusion. The original comment is now outdated and slightly redundant IMO so I'll simplify it. Let me know if there's anything else in particular you'd like me to add/keep.